File: server/app/models/session.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base
from .base import LOCALTIME


SESSION_STATUS_STREAMING = 0
SESSION_STATUS_IDLE = 1


class Session(Base):
    __tablename__ = "sessions"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False, default='')
    agent_id = Column(Integer, nullable=False, index=True)
    status = Column(Integer, default=SESSION_STATUS_IDLE, nullable=False)
    created_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=datetime.now, server_default=LOCALTIME, onupdate=datetime.now, nullable=False)

    messages = relationship(
        "Message",
        back_populates="session",
        primaryjoin="Session.id == Message.session_id",
        foreign_keys="Message.session_id"
    )

File: server/internal/model/session.go
package model

import "time"

// Session status constants.
const (
	SessionStatusStreaming = 0 // Session is currently processing a message
	SessionStatusIdle      = 1 // Session is idle and ready for new messages
)

// Session represents a conversation session between a user and an agent.
// Each session belongs to one agent and contains a series of messages.
type Session struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Title     string    `gorm:"type:varchar(255);not null;default:''" json:"title"`
	AgentID   int64     `gorm:"not null;index;column:agent_id" json:"agent_id"`
	Status    int       `gorm:"not null;default:1" json:"status"`
	CreatedAt time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
	UpdatedAt time.Time `gorm:"not null;autoUpdateTime" json:"updated_at"`
}

func (Session) TableName() string { return "sessions" }
